src/premiscale/hypervisor/__init__.py

```diff
@@ -19,12 +19,14 @@
 log = logging.getLogger(__name__)
 
 
-def build_hypervisor_connection(host: Host) -> Libvirt:
+def build_hypervisor_connection(host: Host, readonly: bool = False) -> Libvirt:
     """
     Build a Libvirt connection object based on the user-provided configuration of the host.
     """
     conf = unstructure(host)
 
+    # Remove the SSH key from the configuration as it's not needed for the connection.
+    # It's only actually needed for the SSH config, which is parsed and produced when the config is validated.
```

Comment:
Just doing some research, it seems these factory methods aren't that bad, after all.

https://stackoverflow.com/a/115853

They're relatively pythonic? I've been struggling with these since I seem to structure my modules and submodules this way a lot in this package, so it's good to see that it's not an anti-pattern insomuch as it gives us the context switch we're looking for with the preprocessing overhead.